packages/table/src/interactions/selectable.tsx

```diff
@@ -256,71 +220,101 @@ export class DragSelectable extends React.Component<IDragSelectableProps, {}> {
             || ignoredSelectors.some((selector: string) => element.closest(selector) != null);
     }
 
-    private getDragSelectedRegions(event: MouseEvent, coords: ICoordinateData) {
-        const { selectedRegions, selectedRegionTransform } = this.props;
+    // Update logic
+    // ============
 
-        let region = this.props.allowMultipleSelection
-            ? this.props.locateDrag(event, coords)
-            : this.props.locateClick(event);
+    private handleUpdateExistingSelection = (selectedRegionIndex: number, event: MouseEvent) => {
+        const { onSelection, selectedRegions } = this.props;
 
-        if (!Regions.isValid(region)) {
-            return undefined;
+        if (DragEvents.isAdditive(event)) {
+            // remove just the clicked region, leaving other selected regions in place
+            const nextSelectedRegions = selectedRegions.slice();
+            nextSelectedRegions.splice(selectedRegionIndex, 1);
+            onSelection(nextSelectedRegions);
```

Comment:
whoops, should use `this.maybeInvokeSelectionCallback` here and elsewhere! will add more unit tests.